lib/rubygems/config_file.rb

```diff
@@ -261,7 +261,12 @@ def check_credentials_permissions
   # Location of RubyGems.org credentials
 
   def credentials_path
-    File.join Gem.user_home, '.gem', 'credentials'
+    credentials = File.join Gem.user_home, '.gem', 'credentials'
+    if File.exist? credentials
+      credentials
+    else
+      File.join Gem.data_home, "gem", "credentials"
+    end
```

Comment:
This pattern of checking file existence appears to occur several times.  Would it be helpful to encapsulate it in a utility method?

```ruby
def find_home_file(filename)
  path = File.join Gem.user_home, ".gem", filename
  if File.exist? path
    path
  else
    File.join Gem.data_home, "gem", filename
  end
end
```